docs/en/middleware.rst

```diff
@@ -42,18 +42,21 @@ inspecting the request object you can configure authentication appropriately::
         $service = new AuthenticationService();
         if (strpos($path, '/api') === 0) {
             // Accept API tokens only
-            $service->loadAuthenticator('Authentication.Token');
-            $service->loadIdentifier('Authentication.Token');
+            $service->loadAuthenticator('Authentication.Token', [
+                'identifier' => 'Authentication.Token',
+            ]);
 
             return $service;
         }
 
         // Web authentication
         // Support sessions and form login.
-        $service->loadAuthenticator('Authentication.Session');
-        $service->loadAuthenticator('Authentication.Form');
-
-        $service->loadIdentifier('Authentication.Password');
+        $service->loadAuthenticator('Authentication.Session', [
+            'identifier' => 'Authentication.Password',
```

Comment:
Will we also support `identifier` being an object? If we're going to change how authentication is wired up, we could make it more compatible with the DI container.

Reply:
Already done, afaik there is a test case added
https://github.com/cakephp/authentication/pull/712/files#diff-b9a4b24bdb898b49129cd5b64f3b2f3cf9e71da882e8f07dc277d3df0d96394fR143